etc/service-bus.api.ts

```diff
@@ -0,0 +1,411 @@
+  // WARNING: The type "MessageSessionOptions" needs to be exported by the package (e.g. added to index.ts)
+  static create(context: ClientEntityContext, options?: MessageSessionOptions): Promise<MessageSession>;
+  getState(): Promise<void>;
+  isOpen(): boolean;
+  maxAutoRenewDurationInSeconds: number;
+  maxConcurrentCallsPerSession?: number;
+  maxConcurrentSessions?: number;
+  maxMessageWaitTimeoutInSeconds: number;
+  peek(messageCount?: number): Promise<ReceivedMessageInfo[]>;
+  peekBySequenceNumber(fromSequenceNumber: Long, messageCount?: number): Promise<ReceivedMessageInfo[]>;
+  // WARNING: The type "SessionHandlerOptions" needs to be exported by the package (e.g. added to index.ts)
+  receive(onSessionMessage: OnSessionMessage, onError: OnError, options?: SessionHandlerOptions): void;
+  receiveBatch(maxMessageCount: number, maxWaitTimeInSeconds?: number): Promise<ServiceBusMessage[]>;
+  receiveDeferredMessage(sequenceNumber: Long): Promise<ServiceBusMessage | undefined>;
+  receiveDeferredMessages(sequenceNumbers: Long[]): Promise<ServiceBusMessage[]>;
+  receiveMode: ReceiveMode;
+  renewLock(): Promise<Date>;
+  sessionId?: string;
+  sessionLockedUntilUtc?: Date;
+  setState(state: any): Promise<void>;
+  // WARNING: The type "DispositionType" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "DispositionOptions" needs to be exported by the package (e.g. added to index.ts)
+  settleMessage(message: ServiceBusMessage, operation: DispositionType, options?: DispositionOptions): Promise<any>;
+}
+
+// @public
+class MessagingError extends Error {
+  constructor(message: string);
+  condition?: string;
+  info?: any;
+  name: string;
+  retryable: boolean;
+  translated: boolean;
+}
+
+// @public
+class Namespace {
+  constructor(config: ConnectionConfig, options?: NamespaceOptions);
+  close(): Promise<any>;
+  // WARNING: The type "NamespaceOptionsBase" needs to be exported by the package (e.g. added to index.ts)
+  static createFromAadTokenCredentials(host: string, credentials: ApplicationTokenCredentials | UserTokenCredentials | DeviceTokenCredentials | MSITokenCredentials, options?: NamespaceOptions): Namespace;
+  static createFromConnectionString(connectionString: string, options?: NamespaceOptions): Namespace;
+  // WARNING: The type "NamespaceOptionsBase" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "NamespaceOptionsBase" needs to be exported by the package (e.g. added to index.ts)
+  static createFromTokenProvider(host: string, tokenProvider: TokenProvider, options?: NamespaceOptionsBase): Namespace;
+  createQueueClient(queueName: string, options?: QueueClientOptions): QueueClient;
+  createSubscriptionClient(topicName: string, subscriptionName: string, options?: SubscriptionClientOptions): SubscriptionClient;
+  createTopicClient(topicName: string): TopicClient;
+  // (undocumented)
```

Comment:
This is correct. It should be documented.

Reply:
I thought I created the docs for this.. but it looks like I didn't do it correctly -- will take a look at that.